workers/unity/Assets/Fps/Scripts/Editor/MapTemplateInspector.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+namespace Fps.Editor
+{
+    [CustomEditor(typeof(MapTemplate))]
+    public class MapTemplateInspector : UnityEditor.Editor
+    {
+        MapTemplate template;
+
+        public void OnEnable()
+        {
+            template = (MapTemplate) target;
+        }
+
+        public override void OnInspectorGUI()
+        {
+            int? elementToRemove = null;
+
+            // Template bitmap asset
+            template.templateBitmap = (Texture2D) EditorGUILayout.ObjectField(new GUIContent(
+                    "Template bitmap",
+                    "2D bitmap to indicate tile types."),
+                template.templateBitmap,
+                typeof(MapTemplate),
+                false);
+
+            // Units per tile
+            template.unitSize = EditorGUILayout.FloatField("Unity Size", template.unitSize);
+
+            // Collection asset
+            template.defaultTileCollection = (TileTypeCollection) EditorGUILayout.ObjectField(
+                new GUIContent("Default Tile Collection"),
+                template.defaultTileCollection,
+                typeof(TileTypeCollection),
+                false);
+
+            EditorGUILayout.LabelField("Tile Collections");
+            EditorGUI.indentLevel++;
+            for (int i = 0; i < template.tileCollections.Length; i++)
+            {
+                // Calculate positioning of the elements on this line
+                var collection = template.tileCollections[i];
+                var color = collection?.DisplayColor ?? Color.clear;
+                var rect = EditorGUILayout.GetControlRect(false, 16);
+                var colorRect = new Rect(rect.x + 16, rect.y, 16, rect.height);
```

Comment:
you use 16 a lot. maybe store it in a field?

Reply:
Its purely to offset by the size of this icon, is it worth making it a field?

Reply:
would be nice in case we need to change the offset. would make it easier